### ISSUE TITLE
Standard Event on every function

### ISSUE THREAD

**Issue by SKeiler** [Microsoft/ALAppExtensions#365]
Hello,

in the recent years, Microsoft has invested a lot of energy in souce code structure. A lot of existing source code was relocated in new functions and sub functions (for example the big CU80-Code-functions was divided in many small functions). 

With 2 standard events in every function like the trigger events, OnBeforeFunction and OnAfterFunction, there would be thousands of new events in NAV and partners would have a lot of possibilities to subscribe this ones for their apps.

And every source code refactoring would create new events automatically.

**Comment by SirBETE** [comment]
Agree, it would be better if it was just part of the platform, instead of (c)AL functions.

And extra: the OnBeforeFunction should have the exact parameters as the function, with an extra Handled boolean.
If it's true, skip the function (MS code).

**Comment by frivadulla** [comment]
> 
> 
> Agree, it would be better if it was just part of the platform, instead of (c)AL functions.
> 
> And extra: the OnBeforeFunction should have the exact parameters as the function, with an extra Handled boolean.
> If it's true, skip the function (MS code).

I completely agree with this, the skip variable would be really helpful.

**Comment by steinarsig** [comment]
This would at least make sense on all functions that are not local.

**Comment by JesperSchulz** [comment]
Thank you very much for your input. Actually, we do have that feature in our feature backlog, to be picked up when time allows. In our current idea, a developer still has to activate the event publishers through a property though, as there may be public functions, we do not want to have extension events for. Unfortunately I cannot tell you when the feature is going to make it into the product - it is prioritized against other important work load. But at some point, you should see something like what you've requested here :-)

**Comment by SKeiler** [comment]
Hi Jesper,

sound's good. Hopefully this point will have a higher priority now because I think it is neccessary to create verticals with extensions :-)

Your aspect, that their may be functions that shouldn't have Events, sounds comprehensible, but I think it would be better if the developer have to deactivate the default events.

**Comment by JesperSchulz** [comment]
Noted! Thanks. Transferring issue to our development tools team, who has this issue in their backlog.
